### PR TITLE
Upgrade pynamodb to 3.2.1 and bump version to 2.0.0

### PR DESCRIPTION
**Pull request by ryan-lane**
This change upgrades pynamodb to 3.2.1, which is a breaking change for anyone using blind credentials. See the changelog for info about upgrading, if you're using blind credentials.

### CODE REVIEW

**Comment by ryan-lane** [comment]
This change needs to be rebased on the 1.12.0 release, when it's merged, so it shouldn't be merged until that rebase.

**Comment by surbhishah** [comment]
:lock:

**Comment by surbhishah** [comment]
:eyes: @ryan-lane @vivianho

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,14 @@
 
 ## 2.0.0
 
+* This is a breaking release, if you're using blind credentials. This change
+  upgrades to using pynamodb 3.2.1. If you're using blind credentials, it's
+  necessary to first upgrade to confidant 1.12.0, run the
```
Comment (vivianho) [inline CHANGELOG.md]: Change from 1.12.0 to 2.0.0

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,6 +4,14 @@
...
+  migrate\_set\_attribute maintenance script, then upgrade to this version.
+  This is due to a breaking change in pynamodb itself, which requires using
+  specific versions of pynamodb to migrate the underlying data.
+
+# 1
```
Comment (vivianho) [inline CHANGELOG.md]: This should be ## 2.0.0

File: CHANGELOG.md [hunk 3]
```diff
@@ -4,6 +4,14 @@
 
 ## 2.0.0
```
Comment (vivianho) [inline CHANGELOG.md]: I think you should change this to 2.1.0 or 3.0.0

Comment (vivianho) [reply CHANGELOG.md]: You'll also need to bump the version in setup.py

File: piptools_requirements.txt [hunk 4]
```diff
@@ -1,5 +1,8 @@
 #
 # This file is autogenerated by pip-compile
+# To update, run:
```
Comment (vivianho) [inline piptools_requirements.txt]: Revert changes to this file, it's not needed.

File: requirements.txt [hunk 5]
```diff
@@ -1,5 +1,8 @@
 #
 # This file is autogenerated by pip-compile
+# To update, run:
```
Comment (vivianho) [inline requirements.txt]: Revert changes to this file, it's not needed.

File: requirements.txt [hunk 6]
```diff
@@ -45,7 +42,7 @@ pep8==1.5.7
 pyasn1==0.1.9
 pycparser==2.17
 pyflakes==1.5.0           # via flake8
-pynamodb==3.2.1
+pynamodb==2.2.0
```
Comment (vivianho) [inline requirements.txt]: @surbhishah this should be 3.2.1.

**Pull request merged**